query_modules/text_search.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2024 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include <mgp.hpp>
+// #include "query/procedure/mg_procedure_impl.hpp"
+// #include "storage/v2/indices/mgcxx_mock.hpp"
+#include "storage/v2/mgcxx_mock.hpp"
+
+namespace TextSearch {
+constexpr std::string_view kProcedureSearch = "search";
+constexpr std::string_view kParameterLabel = "label";
+constexpr std::string_view kParameterSearchString = "search_query";
+constexpr std::string_view kReturnNode = "node";
+
+void Search(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory);
+}  // namespace TextSearch
+
+void Search(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::MemoryDispatcherGuard guard{memory};
+  const auto record_factory = mgp::RecordFactory(result);
+  auto arguments = mgp::List(args);
+  auto label = arguments[0].ValueString();
+  auto search_query = arguments[1].ValueString();
+
+  // 1. See if the given label is text-indexed
+  if (!mgp::graph_has_text_index(memgraph_graph, label.data())) {
+    return;
```

Comment:
Why does this not log or send a message to the user?

Reply:
Of course, the module should definitely let the user know about this. I’m currently ironing out how Memgraph interacts with Tantivy, but I’ll resolve this ASAP.

Reply:
Update: done in the meantime